integration/build/build_test.go

```diff
@@ -22,7 +22,7 @@ import (
 )
 
 func TestBuildWithRemoveAndForceRemove(t *testing.T) {
-	defer setupTest(t)()
+	t.Cleanup(setupTest(t))
```

Comment:
I considered it at first, but thought that it might actually be desirable in some cases to run the cleanup in defer instead of the t.Cleanup.

But overall, I agree that having `t.Cleanup` inside the `setupTest` and freeing the caller from the responsibility for handling it would be more convenient.

Reply:
Do we have a case where we need it in `defer`? Tempted to say that could be a ["YAGNI"](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it#:~:text=%22You%20aren%27t%20gonna%20need,add%20functionality%20until%20deemed%20necessary.).

Reply:
I didn't find it, but I didn't really review all the tests. It was definitely easier to find all tests that have `setupTest()` -> `t.Run()` -> `t.Parallel()` call chain than to review what every test with `defer setupTest(t)()` does 😄